Added documentation for AI Data Mapper (#968)
wso2/docs-mi: en/docs/reference/mediators/data-mapper-mediator.md

- **match** – check whether the input match with a (JS) Regular Expression

## AI Data Mapper

> The MI Copilot AI Assistant will allow you to seamlessly generate the input-output mapping. Simply load the input and output schema (of any type) to the relevant sections as shown below and click **Map**. A **Clear** button is also available for convenience. 

Reviewer: Instead of `also available for convenience`, shall we mention what the clear btn does?

> The MI Copilot AI Assistant will allow you to seamlessly generate the input-output mapping. Simply load the input and output schema (of any type) to the relevant sections as shown below and click **Map**. A **Clear** button is also available to clear all mappings. 

Reviewer: Shall we reword the second line: 

Use the **Clear** button to clear all mappings.

!!! info "WSO2 Account and Usage Limits"

>     To access this feature, you need to create a WSO2 account. If you have not created an account, accessing copilot or using the mapping feature will prompt you to do so. If you exceed your account's usage limit, it will be refreshed after the specified time limit.

Reviewer: ```suggestion
    To use this feature, you must have a WSO2 account. If you do not have one, attempting to access Copilot or the mapping feature will prompt you to create an account. If your account exceeds the usage limit, it will be refreshed after the specified time limit.
```

!!! info "Review and Fine-tune"

>     Since AI powers the automated mapping, it may contain errors. Always verify the accuracy of the generated mappings using the `</>` button to access the TypeScript (TS) file and fine-tune them.

Reviewer 2: ```suggestion
    Since AI powers the automated mapping, it may contain errors. Always verify the accuracy of the generated mappings. To fine tune the mappings you can use the `</>` button and edit the TypeScript (TS) file.
```

>     Please note that since automated mapping is powered by AI, some mistakes or surprises may occur. It is always recommended to confirm the accuracy of the generated mappings using the </> TypeScript (TS) file and fine-tune them.

Reviewer: ```suggestion
    Since AI powers the automated mapping, it may contain errors. Always verify the accuracy of the generated mappings using the </> TypeScript (TS) file and fine-tune them.
```
